test/test.js

```diff
@@ -282,4 +282,62 @@ describe('Chinese calendar', function() {
         expect(chineseCalendar.leapYear(chineseDate.year())).toEqual(true);
         expect(chineseDate.formatDate()).toEqual("1917/05/05");
     });
+
+    it('should include intercalary months when incrementing months', function() {
+        var testCases = [
+            "1895/05/05",
+            "1914/05/05",
+            "1995/08/01",
+            "1998/05/01",
+            "2014/09/02",
+        ];
+        testCases.forEach(function(testCase) {
+            var chineseDate = chineseCalendar.parseDate(null, testCase);
+            var year = chineseDate.year();
+            var intercalaryMonth = chineseCalendar.intercalaryMonth(year);
+            var monthIndex = chineseDate.month();
+            var monthString = chineseDate.formatDate("m");
+
+            // initial date isn't an intercalary month
+            expect(monthIndex).not.toEqual(intercalaryMonth);
+            expect(monthString.charAt(monthString.length - 1))
+                .not.toEqual("i");
+
+            // next month is an intercalary month
+            chineseDate.add(1, 'm');
+            var nextDateYear = chineseDate.year();
+            var nextDateMonthIndex = chineseDate.month();
+            var nextDateMonthString = chineseDate.formatDate("m");
+            expect(nextDateYear).toEqual(year);
+            expect(nextDateMonthIndex).toEqual(monthIndex + 1);
+            expect(nextDateMonthIndex).toEqual(intercalaryMonth);
+            expect(nextDateMonthString.charAt(nextDateMonthString.length - 1))
+                .toEqual("i");
+
+            // and next month isn't
+            chineseDate.add(1, 'm');
+            nextDateYear = chineseDate.year();
+            nextDateMonthIndex = chineseDate.month();
+            nextDateMonthString = chineseDate.formatDate("m");
+            expect(nextDateYear).toEqual(year);
+            expect(nextDateMonthIndex).toEqual(monthIndex + 2);
+            expect(nextDateMonthString.charAt(nextDateMonthString.length - 1))
+                .not.toEqual("i");
+        });
+    });
+
+    it('should format dates in the first month correctly', function() {
+        for(var year = 1888; year < 2112; year++) {
+            var newYearString = year + "/01/01";
+            var newYear = chineseCalendar.parseDate(null,newYearString); 
+            expect(newYear.formatDate()).toBe(newYearString);
+
+            var intercalaryMonth = chineseCalendar.intercalaryMonth(year);
+            if (intercalaryMonth === 1) {
```

Comment:
Not in the range we currently cover (1888-2111), but [Aslaksen's paper (Table 23)](http://www.math.nus.edu.sg/aslaksen/calendar/cal.pdf) quotes 6 occurrences between 1645 and 2644.

---

Note to my future me: there are no intercalary 12th months between 1645 and 2644.